src/useCookieConsent.ts

```diff
@@ -16,19 +16,27 @@ export const useCookieConsent = (
 ): CookieConsentHookState => {
```

Comment:
Here, the `CookieConsentHookState` type contains some functions that include the options for the `js-cookie` package. For example, if you use `acceptCookies` you can pass `options?: AcceptCookiesOptions` that would set the options on the underlying `js-cookie` package. This doesn't make sense to have when you're only saving consent into the `localStorage` or any other objects implementing Storage API.

This means that this return type should be generic in some way to allow making the return type not include the `options` for `js-cookie`. I'd say it's worth reading up on "Typescript Generics" and "Conditional Types".